docs/scripts/build_examples.py

```diff
@@ -0,0 +1,241 @@
+"""Utility for creating the examples."""
+
+import argparse
+import os
+import pathlib
+import shutil
+import textwrap
+from subprocess import DEVNULL, STDOUT, check_call
+
+from tqdm import tqdm
+
+from baybe.telemetry import VARNAME_TELEMETRY_ENABLED
+
+parser = argparse.ArgumentParser()
+parser.add_argument(
+    "-e",
+    "--ignore_examples",
+    help="Ignore the examples by not executing them.",
+    action="store_true",
+)
+parser.add_argument(
+    "-w",
+    "--include_warnings",
+    help="Include warnings when processing the examples. The default is ignoring them.",
+    action="store_true",
+)
+parser.add_argument(
+    "-f",
+    "--force",
+    help="Force-build the documentation, even when there are warnings or errors.",
+    action="store_true",
+)
+
+# Parse input arguments
+args = parser.parse_args()
+IGNORE_EXAMPLES = args.ignore_examples
+INCLUDE_WARNINGS = args.include_warnings
+FORCE = args.force
+
+
+def build_examples(example_dest_dir: str, ignore_examples: bool):
+    """Create the documentation version of the examples files.
+
+    Note that this deletes the destination directory if it already exists.
+
+    Args:
+        example_dest_dir: The destination directory.
+        ignore_examples: A flag denoting whether or not to ignore the examples. If set,
+            then the examples are constructed, but not executed.
+    """
+    # Folder where the .md files created are stored
+    examples_directory = pathlib.Path(example_dest_dir)
+
+    # if the destination directory already exists it is deleted
```

Comment:
What is the rationale here? Don't we want to reuse when it already exists?

Reply:
Well, if you call this script, you want to rebuild the examples. Also, since there is the conversion to Jupyter Notebooks, I am not sure if `sphinx` would even be able to properly detect that stuff did not change.

Reply:
OK, I see. I think the confusion lies a bit in the fact how the commands are currently assembled. Right now, if you want to repeatedly build the doc in the fastest way, it requires to switch commands to avoid rebuilding the cache, i.e. we first need to use `docs` and then `docs-quickbuild`, and quickbuild fails if used as a first command. I think this design is a bit in contrast to how most caching mechanisms work (also the one from sphinx). From a user perspective, I think the easier approach would be if we could just always use `docs` and it would automatically do only the necessary work in each command (just like sphinx does it), and instead provide a flag if we *explicitly* want to bypass caches.

Reply:
Ok, since most of the open points are boilerplate, let me resolve this discussion.